Automattic/newspack-newsletters
# fix(editor): render notice on sync error

### All Submissions:






### Changes proposed in this Pull Request:

1200530782742699-as-1207981078972791

Creates an endpoint to retrieve the sync error message to be fetched once the editor has finished saving a post.

This PR also cleans up some unused react hooks from the newsletter editor.

### How to test the changes in this Pull Request:

1. Configure the plugin to use Mailchimp
2. Draft a new newsletter, save and confirm the draft is synced to Mailchimp
3. Add the following to line `784` of `includes/service-providers/mailchimp/class-newspack-newsletters-mailchimp.php`

```php
			throw new Exception( 'Test Error.' );
```

4. Save the draft campaign again and confirm the notice appears after saving
5. Refresh the page and confirm the notice doesn't render
6. Save again and confirm it reappears
7. Without refreshing the editor, remove the added line, make changes to the draft and save once again
8. Confirm the notice disappears after saving and the changes are synced to Mailchimp

### Other information:
